### PR TITLE
EVEREST-496 Add multi-namespaces support to DBClusters

### PR DESCRIPTION
**Pull request by recharte**
[EVEREST-496](https://jira.percona.com/browse/EVEREST-496) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**Add multi-namespaces support to DBClusters**
---
**NOTE FOR REVIEWERS: Still missing api-tests,  I'll submit a follow-up PR to address the integration tests once the dependencies are sorted out.**
**Problem:**
EVEREST-496

Add the namespace prefix to every DB related endpoint

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?

**Tests**
- [ ] Is an Integration test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

[EVEREST-496]: https://perconadev.atlassian.net/browse/EVEREST-496?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/database_cluster_restore.go [hunk 1]
```diff
@@ -79,33 +79,33 @@ func (e *EverestServer) CreateDatabaseClusterRestore(ctx echo.Context) error {
...
 	}
 
-	return e.proxyKubernetes(ctx, "", databaseClusterRestoreKind, "")
+	return e.proxyKubernetes(ctx, namespace, databaseClusterRestoreKind, "")
 }
 
 // DeleteDatabaseClusterRestore Delete the specified cluster restore on the specified kubernetes cluster.
-func (e *EverestServer) DeleteDatabaseClusterRestore(ctx echo.Context, name string) error {
-	return e.proxyKubernetes(ctx, "", databaseClusterRestoreKind, name)
+func (e *EverestServer) DeleteDatabaseClusterRestore(ctx echo.Context, namespace string, name string) error {
```
Comment (oksana-grishchenko) [inline api/database_cluster_restore.go:86]:
```suggestion
func (e *EverestServer) DeleteDatabaseClusterRestore(ctx echo.Context, namespace, name string) error {
```
There are many of such in the PR, not a big deal, it's just triggering my OCD😅

File: deploy/quickstart-k8s.yaml [hunk 2]
```diff
@@ -23,6 +23,9 @@ kind: ClusterRole
 metadata:
   name: everest-admin-cluster-role
 rules:
+  - apiGroups: ["everest.percona.com"]
+    resources: ["*"]
+    verbs: ["*"]
```
Comment (oksana-grishchenko) [inline deploy/quickstart-k8s.yaml:28]: A bit concerned about the broad permissions, why don't we restrict it to the necessary list only?

**Review by oksana-grishchenko — APPROVED** [review]

File: api/database_cluster_restore.go (hunk 1 above)
Comment (recharte) [reply api/database_cluster_restore.go]: Fixed :stuck_out_tongue: I can't fix the autogenerated code but I think everything else got fixed.

File: deploy/quickstart-k8s.yaml (hunk 2 above)
Comment (recharte) [reply deploy/quickstart-k8s.yaml]: Done

**Review by oksana-grishchenko — APPROVED** [review]

**Pull request merged**